polkadot/node/core/prospective-parachains/src/fragment_chain/mod.rs

```diff
@@ -65,134 +80,171 @@
 //!      resolved by having candidates reference their parent by candidate hash.
 //!
 //! However, dealing with cycles increases complexity during the backing/inclusion process for no
-//! practical reason. Therefore, fragment chains will not accept such candidates.
+//! practical reason.
+//! These cycles may be accepted by fragment chains while candidates are part of the unconnected
+//! storage, but they will definitely not make it to the best chain.
 //!
 //! On the other hand, enforcing that a parachain will NEVER be acyclic would be very complicated
 //! (looping through the entire parachain's history on every new candidate or changing the candidate
 //! receipt to reference the parent's candidate hash).
 //!
+//! Therefore, we don't provide a guarantee that a cycle-producing parachain will work (although in
+//! practice they probably will if the cycle length is larger than the number of assigned cores
+//! multiplied by two).
+//!
 //! # Spam protection
 //!
-//! As long as the [`CandidateStorage`] has bounded input on the number of candidates supplied,
-//! [`FragmentChain`] complexity is bounded. This means that higher-level code needs to be selective
-//! about limiting the amount of candidates that are considered.
+//! As long as the supplied number of candidates is bounded, [`FragmentChain`] complexity is
+//! bounded. This means that higher-level code needs to be selective about limiting the amount of
+//! candidates that are considered.
+//!
+//! Practically speaking, the collator-protocol will not allow more than `max_candidate_depth + 1`
+//! collations to be fetched at a relay parent and statement-distribution will not allow more than
+//! `max_candidate_depth + 1` seconded candidates at a relay parent per each validator in the
+//! backing group. Considering the `allowed_ancestry_len` configuration value, the number of
+//! candidates in a `FragmentChain` (including its unconnected storage) should not exceed:
+//!
+//! `allowed_ancestry_len * (max_candidate_depth + 1) * backing_group_size`.
 //!
 //! The code in this module is not designed for speed or efficiency, but conceptual simplicity.
 //! Our assumption is that the amount of candidates and parachains we consider will be reasonably
 //! bounded and in practice will not exceed a few thousand at any time. This naive implementation
 //! will still perform fairly well under these conditions, despite being somewhat wasteful of
 //! memory.
+//!
+//! Still, the expensive candidate data (CandidateCommitments) are wrapped in an `Arc` and shared
```

Comment:
Overseer requires the main subsystem future to be `Send`. The prospective-parachains is not `Send` if we use an `Rc` here, since this `Rc` is being held across `await` points in the code.